pkg/pool-manager/mac-pool-map.go

```diff
@@ -0,0 +1,118 @@
+package pool_manager
+
+import (
+	kubevirt "kubevirt.io/client-go/api/v1"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// isMacPendingCommit checks mac entry if an update is needed
+func (m macMap) isMacPendingCommit(macAddress string) bool {
+	macEntry, exist := m.findByMacAddress(macAddress)
+	if exist {
+		if macEntry.isPending() {
+			return true
+		}
+	}
+	return false
+}
+
+// clearMacTransactionFromMacEntry clears mac entry's transactionTimestamp, signalling that no further update is needed
+func (m *macMap) clearMacTransactionFromMacEntry(macAddress string) {
+	macEntry, exist := m.findByMacAddress(macAddress)
+	if exist {
+		(*m)[macAddress] = macEntry.resetTimestamp()
+	}
+}
+
+func (m macMap) findByMacAddress(macAddress string) (macEntry, bool) {
+	macEntry, exist := m[macAddress]
+	return macEntry, exist
+}
+
+// removeMacEntry deletes a macEntry from macPoolMap
+func (m *macMap) removeMacEntry(macAddress string) {
+	delete(*m, macAddress)
+}
+
+// filterInByInstanceName creates a subset map from macPoolMap, holding only macs that belongs to a specific instance (pod/vm)
+func (m macMap) filterInByInstanceName(instanceName string) (map[string]macEntry, error) {
+	instanceMacMap := map[string]macEntry{}
+
+	for macAddress, macEntry := range m {
+		if macEntry.instanceName == instanceName {
+			instanceMacMap[macAddress] = macEntry
+		}
+	}
+	return instanceMacMap, nil
+}
+
+func (m *macMap) createOrUpdateEntry(macAddress, instanceFullName, macInstanceKey string) {
+	(*m)[macAddress] = macEntry{}.createOrUpdate(instanceFullName, macInstanceKey, nil)
+}
+
+// isMacReadyForTransactionUpdate checks if the transaction TS recorded on the mac entry has passed the last
+// timestamp annotation that persisted in the mac yaml, signalling that we need to update this entry.
+func (m macMap) isMacReadyForTransactionUpdate(macAddress string, lastPersistentTime *time.Time) (bool, error) {
+	macEntry, exist := m.findByMacAddress(macAddress)
+	if !exist {
+		return false, errors.New("mac address %s not found in macPoolMap")
+	}
+
+	if macEntry.transactionTimestamp.After(*lastPersistentTime) {
+		return false, nil
+	}
+	return true, nil
+}
+
+// isMacTransactionAfterTime checks if mac entry has not been handled for more than a given time in seconds
+func (m macMap) isMacTransactionAfterTime(macAddress string, waitTime int) (bool, error) {
+	macEntry, exist := m.findByMacAddress(macAddress)
+	if !exist {
+		return false, nil
+	}
+	if macEntry.isPending() {
+		macTransactionExpiration := macEntry.addTime(time.Duration(waitTime) * time.Second)
+		if now().After(macTransactionExpiration) {
+			return true, nil
+		}
+	}
```

Comment:
Move all this to a `hasExpiredTransaction` function into `macentry.go` so we do the following here.
```golang
return macEntry.hasExpiredTransaction(waitTime)
```

Reply:
Done. Like!